app/middlewares/device.ts

```diff
@@ -0,0 +1,37 @@
+import * as restify from 'restify';
+import { IRequest, IResponse } from '../interfaces/utils/IServer';
+import { IDevice } from '../interfaces/models';
+import { UnauthorizedError, InternalServerError } from 'restify-errors';
+import IMiddleware from '../interfaces/utils/IMiddleware';
+import Device from '../models/device.model';
+
+export default class AuthDeviceCheckMiddleware implements IMiddleware {
+  public async init(req: IRequest, res: IResponse, next: restify.Next) {
+    const device = <IDevice> await Device.findOne({
+      where: {
+        client_id: req.client_id,
+        user_id: req.body.user_id
+      },
+    });
+
+    if (!device) {
+      return next();
+    }
+
+    const UUID: string = <string>(req.headers.UUID || req.headers.uuid);
+
+    if (!UUID) {
+      return next(new UnauthorizedError({
+        message: 'No Device UUID was specified in the Request Headers'
+      }));
+    }
+
+    if (UUID !== device.UUID) {
```

Comment:
Since there will be `devices` array instead of only 1 device. You should find it inside an array.